### PR TITLE
6690 expose task id to enketo

### PR DESCRIPTION
**Pull request by njogz**
# Description

Expose task ID to Enketo

medic/cht-core#6690

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Review by dianabarsan — CHANGES_REQUESTED** [review]
Nice work. I have a recommendation about where to place the hydration code. 
Also, please update the documentation here: https://docs.communityhealthtoolkit.org/apps/tutorials/app-forms/#1-define-xls-surveyform-fields and here: https://docs.communityhealthtoolkit.org/apps/reference/forms/app/#xlsform

**Comment by njogz** [comment]
@dianabarsan please have another look.

File: webapp/src/ts/modules/tasks/tasks-content.component.ts [hunk 1]
```diff
@@ -144,7 +144,7 @@ export class TasksContentComponent implements OnInit, OnDestroy, AfterViewInit {
     this.globalActions.settingSelected();
 
     return this
-      .hydrateTaskEmission(task)
+      .hydrateTaskEmission(task, id)
```
Comment (dianabarsan) [inline webapp/src/ts/modules/tasks/tasks-content.component.ts:147]: Can't you take the id from the task? Why the extra param?

Comment (njogz) [reply webapp/src/ts/modules/tasks/tasks-content.component.ts]: Good catch! Now getting it from the task.

File: webapp/tests/karma/ts/modules/tasks/tasks-content.component.spec.ts [hunk 2]
```diff
@@ -472,6 +476,8 @@ describe('TasksContentComponent', () => {
 
       sinon.resetHistory();
       sinon.stub(GlobalActions.prototype, 'setEnketoError');
+      store.overrideSelector(Selectors.getSelectedTask, { _id: 'task-id'});
```
Comment (dianabarsan) [inline webapp/tests/karma/ts/modules/tasks/tasks-content.component.spec.ts:479]: I don't think this is needed anymore, right?

Comment (njogz) [reply webapp/tests/karma/ts/modules/tasks/tasks-content.component.spec.ts:479]: That's right.

**Review by dianabarsan — APPROVED** [review]
Nice work! 

Please update the documentation here: https://docs.communityhealthtoolkit.org/apps/tutorials/app-forms/#1-define-xls-surveyform-fields and here: https://docs.communityhealthtoolkit.org/apps/reference/forms/app/#xlsform

**Pull request merged**